athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/common/PartitionUtil.java

```diff
@@ -0,0 +1,212 @@
+        String[] regExParts = folderNameRegEx.split("/");
+        String[] folderParts = folderModel.split("/");
+        if (folderParts.length >= regExParts.length) {
+            for (int i = 0; i < regExParts.length; i++) {
+                Matcher matcher = Pattern.compile(regExParts[i]).matcher(folderParts[i]);
+                if (matcher.matches() && matcher.groupCount() > 0) {
+                    String partitionColumn;
+                    String columnValue;
+                    if (matcher.group(0).equals(matcher.group(1))) { // Non-hive partition (e.g, /{partitionKey1}//{partitionKey2})
+                        Matcher nonHivePartitionPatternMatcher = PARTITION_PATTERN.matcher(partitionPatternParts[i]);
+                        if (nonHivePartitionPatternMatcher.matches()) {
+                            partitionColumn = nonHivePartitionPatternMatcher.group(1);
+                        }
+                        else { // unknown partition layout
+                            throw new IllegalArgumentException("Unsupported partition layout pattern. partition.pattern is " + partitionPattern);
+                        }
+                    }
+                    else if (matcher.group(0).contains("=")) { // Hive partition (e.g. /folderName1={partitionKey1}/folderName2={partitionKey2})
+                        partitionColumn = matcher.group(0).substring(0, matcher.group(0).indexOf("="));
+                    }
+                    else { // mixed partition (e.g, /folderName{partitionKey})
+                        Matcher mixedPartitionPatternMatcher = PARTITION_PATTERN.matcher(partitionPatternParts[i]);
+                        if (mixedPartitionPatternMatcher.find()) {
+                            partitionColumn = mixedPartitionPatternMatcher.group(1);
+                        }
+                        else { // unknown partition layout
+                            throw new IllegalArgumentException("Unsupported partition layout pattern. partition.pattern is " + partitionPattern);
+                        }
+                    }
+                    columnValue = matcher.group(1);
+                    Optional<PartitionColumnData> optionalStoragePartition = produceStoragePartition(partitionColumns, partitionColumn, columnValue);
+                    optionalStoragePartition.ifPresent(partitions::add);
+                }
+            }
+        }
+        return partitions;
+    }
+
+    // helpers
+    /**
+     * Validates partition column types. As of now, only VARCHAR (string or varchar in Glue Table)
+     * @param columns List of Glue Table's columns
+     */
+    private static void validatePartitionColumnTypes(List<Column> columns)
+    {
+        for (Column column : columns) {
+            String columnType = column.getType();
+            LOGGER.info("validatePartitionColumnTypes - Field type of {} is {}", column.getName(), columnType);
+            switch (columnType) {
```

Comment:
You should check to see if this could ever be mixed case. If so you should probably lowercase the `columnType` before doing the switch.